pkg/suggestion/nasrl_service.py

```diff
@@ -1,22 +1,271 @@
+
+        arc = arc.tolist()
+        organized_arc = [0 for _ in range(self.num_layers)]
+        record = 0
+        for l in range(self.num_layers):
+            organized_arc[l] = arc[record: record + l + 1]
+            record += l + 1
+
+        nn_config = dict()
+        nn_config['num_layers'] = self.num_layers
+        nn_config['input_size'] = self.input_size
+        nn_config['output_size'] = self.output_size
+        nn_config['embedding'] = dict()
+        for l in range(self.num_layers):
+            opt = organized_arc[l][0]
+            nn_config['embedding'][opt] = self.search_space[opt].get_dict()
+
+        organized_arc_json = json.dumps(organized_arc)
+        nn_config_json = json.dumps(nn_config)
+
+        organized_arc_str = str(organized_arc_json).replace('\"', '\'')
+        nn_config_str = str(nn_config_json).replace('\"', '\'')
+
+        self.logger.info("\nNew Neural Network Architecture (internal representation):")
+        self.logger.info(organized_arc_json)
+        self.logger.info("\nCorresponding Seach Space Description:")
+        self.logger.info(nn_config_str)
+        self.logger.info("")
+
         trials = []
+        trials.append(api_pb2.Trial(
+                study_id=request.study_id,
+                parameter_set=[
+                    api_pb2.Parameter(
+                        name="architecture",
+                        value=organized_arc_str,
+                        parameter_type= api_pb2.CATEGORICAL),
+                    api_pb2.Parameter(
+                        name="nn_config",
+                        value=nn_config_str,
+                        parameter_type= api_pb2.CATEGORICAL)
+                ], 
+            )
+        )
 
+        channel = grpc.beta.implementations.insecure_channel(self.manager_addr, self.manager_port)
+        with api_pb2.beta_create_Manager_stub(channel) as client:
+            for i, t in enumerate(trials):
+                ctrep = client.CreateTrial(api_pb2.CreateTrialRequest(trial=t), 10)
+                trials[i].trial_id = ctrep.trial_id
+            self.logger.info("Trial {} Created\n".format(ctrep.trial_id))
+            self.prev_trial_id = ctrep.trial_id
+        
+        self.ctrl_step += 1
         return api_pb2.GetSuggestionsReply(trials=trials)
+
+    def GetEvaluationResult(self, studyID):
+        worker_list = []
+        channel = grpc.beta.implementations.insecure_channel(self.manager_addr, self.manager_port)
+        with api_pb2.beta_create_Manager_stub(channel) as client:
+            gwfrep = client.GetWorkerFullInfo(api_pb2.GetWorkerFullInfoRequest(study_id=studyID, trial_id=self.prev_trial_id, only_latest_log=True), 10)
+            worker_list = gwfrep.worker_full_infos
+
+        completed_count = 0
+        for w in worker_list:
+            if w.Worker.status == api_pb2.COMPLETED:
+                for ml in w.metrics_logs:
+                    if ml.name != self.objective_name:
```

Comment:
if w.metrics_logs is like [objectivemetrics, othermetrics], line 214 will return othermetrics value (ml is othermetrics)

how about
```
if ml.name == self.objective_name:
    self.logger.info("Evaluation result of previous candidate: {}".format(ml.values[-1].value))
    return float(ml.values[-1].value)
```
and keep below to keep same indent with line 204 (completed_count can be removed)
```
self.logger.warning("Error. No trial has completed.")
return None
```

Reply:
I believe w.metrics_logs will only return the selected metrics defined in https://github.com/kubeflow/katib/blob/master/examples/nasjob-example-RL.yaml#L12. The metrics collector will parse the pattern <metrics_name>: <metrics_value> from the log of training container and give it to suggestion.

Currently, the algorithm only spawns one trial each time and uses the metrics of the very last trials to update. That is why I will go through all the available metrics and index the last one.

Reply:
w.metrics_logs will return Collection of https://github.com/kubeflow/katib/blob/master/examples/nasjob-example-RL.yaml#L12 and https://github.com/kubeflow/katib/blob/master/examples/nasjob-example-RL.yaml#L15. Maybe you can take a look by debug to verify it

Reply:
Yes, you are right! 
The program has been modified as suggested. Thank you for your reminder!